src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping.hs

```diff
@@ -928,33 +935,42 @@ checkFunction ::
   Function 'Parsed ->
   Sem r (Function 'Scoped)
 checkFunction Function {..} = do
-  funParameter' <- checkParam
-  let scoped = case funParameter' ^. paramName of
-        Nothing -> id
-        Just s -> withBindLocalVariable (LocalVariable s)
+  funParameters' <- checkParams
+  let scoped =
+        foldr
+          ( \param acc -> case param of
```

Comment:
personal preference:
In general I don't like lambda functions unless they are the last argument. I think they make the code less readiable.
I'd prefer if the lambda was defined in a let/where

Reply:
Done